core/src/main/java/org/elasticsearch/search/aggregations/bucket/range/RangeAggregator.java

```diff
@@ -151,7 +154,18 @@ public static Range fromXContent(XContentParser parser) throws IOException {
                         toAsStr = parser.text();
                     } else if (KEY_FIELD.match(currentFieldName)) {
                         key = parser.text();
+                    } else {
+                        XContentParserUtils.throwUnknownField(currentFieldName, parser.getTokenLocation());
                     }
+                } else if (token == XContentParser.Token.VALUE_NULL) {
+                    if (FROM_FIELD.match(currentFieldName) || TO_FIELD.match(currentFieldName)
+                            || KEY_FIELD.match(currentFieldName)) {
+                        // ignore null value
+                    } else if (KEY_FIELD.match(currentFieldName)) {
+                        XContentParserUtils.throwUnknownField(currentFieldName, parser.getTokenLocation());
```

Comment:
is this condition not matched in the condition above? Maybe this should just be an else?